packages/medusa/src/services/product-variant.ts

```diff
@@ -571,10 +630,39 @@ class ProductVariantService extends BaseService {
         raw.map((i) => i.id),
         query.withDeleted ?? false
       )
+      if (relations?.length && relations?.indexOf("prices") > -1) {
```

Comment:
yeah, good idea. I have implemented this everywhere we use the price selection strategy. 

As for how to avoid modifying the api level array for now I create a copy and override the configuration `relations` field. I'm open for better ideas.